examples/scale/frontend/frontend.go

```diff
@@ -60,73 +60,89 @@ func run(cfg config.View) {
 	}
 	fe := pb.NewFrontendServiceClient(conn)
 
-	w := logger.Writer()
-	defer w.Close()
-
 	ticketQPS := int(activeScenario.FrontendTicketCreatedQPS)
 	ticketTotal := activeScenario.FrontendTotalTicketsToCreate
 
-	for {
-		currentCreated := int(atomic.LoadUint32(&totalCreated))
-		if ticketTotal != -1 && currentCreated >= ticketTotal {
-			break
+	totalCreated := 0
+
+	for range time.Tick(time.Second) {
+		for i := 0; i < ticketQPS; i++ {
+			if ticketTotal == -1 || totalCreated < ticketTotal {
+				go runner(fe)
+			}
 		}
+	}
+}
 
-		// Each inner loop creates TicketCreatedQPS tickets
-		var ticketPerRoutine, ticketModRoutine int
-		start := time.Now()
+func runner(fe pb.FrontendServiceClient) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
 
-		if ticketTotal == -1 || currentCreated+ticketQPS <= ticketTotal {
-			ticketPerRoutine = ticketQPS / numOfRoutineCreate
-			ticketModRoutine = ticketQPS % numOfRoutineCreate
-		} else {
-			ticketPerRoutine = (ticketTotal - currentCreated) / numOfRoutineCreate
-			ticketModRoutine = (ticketTotal - currentCreated) % numOfRoutineCreate
-		}
+	g := stateGauge{}
+	defer g.stop()
 
-		var wg sync.WaitGroup
-		for i := 0; i < numOfRoutineCreate; i++ {
-			wg.Add(1)
-			if i < ticketModRoutine {
-				go createPerCycle(&wg, fe, ticketPerRoutine+1, start)
-			} else {
-				go createPerCycle(&wg, fe, ticketPerRoutine, start)
-			}
-		}
+	g.start(mRunnersWaiting)
+	time.Sleep(time.Duration(rand.Int63n(int64(time.Second))))
```

Comment:
Please add a comment (either on this line or function or just header on the file) stating why the sleep exists and what it does (PR description covers that really well - but i think it is important enough to have it in context)